File: aasexplorer-core/src/main/java/com/ficture7/aasexplorer/model/QuestionPaper.java
package com.ficture7.aasexplorer.model;

import static com.ficture7.aasexplorer.util.ObjectUtil.checkNotNull;

/**
 * Represents a question paper resource.
 *
 * @author FICTURE7
 */
public class QuestionPaper extends Resource {

    private final Session session;
    private final int number;

    /**
     * Constructs a new instance of the {@link QuestionPaper} class with the specified resource name
     * and session.
     *
     * @param name    Resource name.
     * @param session Session.
     * @param number  Paper number.
     * @throws NullPointerException {@code name} is null.
     * @throws NullPointerException {@code session} is null.
     */
    public QuestionPaper(String name, Session session, int number) {
        super(name);

        this.session = checkNotNull(session, "session");
        this.number = number;
    }

    /**
     * Gets the {@link Session} of the {@link QuestionPaper}.
     *
     * @return {@link Session} of the {@link QuestionPaper}.
     */
    public Session session() {
        return session;
    }

    /**
     * Gets the paper number of the {@link QuestionPaper}.
     *
     * @return Paper number of the {@link QuestionPaper}.
     */
    public int number() {
        return number;
    }
}

File: aasexplorer-core/src/main/java/com/ficture7/aasexplorer/model/QuestionPaper.kt
package com.ficture7.aasexplorer.model

/**
 * Represents a question paper resource.
 *
 * @author FICTURE7
 */
class QuestionPaper : Resource {

    /**
     * Constructs a new instance of the [QuestionPaper] class with the specified resource name,
     * session and paper number.
     *
     * @param name    Resource name.
     * @param session Session.
     * @param number  Paper number.
     */
    constructor(name: String, session: Session, number: Int) : super(name) {
        this.session = session
        this.number = number
    }

    /**
     * Returns the [Session] of the [QuestionPaper].
     *
     * @return [Session] of the [QuestionPaper].
     */
    val session: Session

    /**
     * Returns the paper number of the [QuestionPaper].
     *
     * @return Paper getNumber of the {@link QuestionPaper}.
     */
    val number: Int
}